.travis.yml

```diff
@@ -46,174 +44,257 @@ cache:
+    compiler: ": #stack 7.10.3"
     addons: {apt: {packages: [libgmp-dev,
                               libmpfr-dev,
                               libmpc-dev,
                               binutils-dev,
                               g++-4.7,
-                              gcc,
-                              ghc-7.10.3],
-                  sources: [hvr-ghc, ubuntu-toolchain-r-test]}}
+                              gcc, ghc-7.10.3], sources: [hvr-ghc, ubuntu-toolchain-r-test]}}
 
   # Nightly builds are allowed to fail
   - env: BUILD=stack ARGS="--resolver nightly"
+    compiler: ": #stack nightly"
     addons: {apt: {packages: [libgmp-dev,
                               libmpfr-dev,
                               libmpc-dev,
                               binutils-dev,
                               g++-4.7,
-                              gcc,
-                              ghc-7.10.3],
-                  sources: [hvr-ghc, ubuntu-toolchain-r-test]}}
+                              gcc, libgmp-dev]}}
 
   # Build on OS X in addition to Linux
+  - env: BUILD=stack ARGS=" "
+    compiler: ": #stack default osx"
+    os: osx
+
   - env: BUILD=stack ARGS="--resolver lts-2"
+    compiler: ": #stack 7.8.4 osx"
     os: osx
 
   - env: BUILD=stack ARGS="--resolver lts-3"
+    compiler: ": #stack 7.10.2 osx"
     os: osx
 
   - env: BUILD=stack ARGS="--resolver lts-5"
+    compiler: ": #stack 7.10.3 osx"
     os: osx
 
   - env: BUILD=stack ARGS="--resolver nightly"
+    compiler: ": #stack nightly osx"
     os: osx
 
   allow_failures:
-  - env: BUILD=cabal GHCVER=head  CABALVER=head
+  - env: BUILD=cabal GHCVER=head  CABALVER=head HAPPYVER=1.19.5 ALEXVER=3.1.7
   - env: BUILD=stack ARGS="--resolver nightly"
 
-before_install:
 
+install:
+# SYMENGINE INSTALL PHASE
+# =======================
+# Download and install symengine
+- |
+  set -ex
+  export SYMENGINE_INSTALL_DIR=$HOME/symengine-install
+  mkdir -p $SYMENGINE_INSTALL_DIR
+  cd $HOME && git clone https://github.com/symengine/symengine.git
+  
+- |
+  if [ `uname` = "Darwin" ]
+  then
+    export CC="clang" && export CXX="clang++" && export CMAKE_C_COMPILER=$CC && export CMAKE_CXX_COMPILER=$CXX
+
+    # build gmp with clang
+    wget https://raw.githubusercontent.com/symengine/dependencies/6a42d290071921a0a478c6883fc0ddd709d664c9/gmp-6.0.0a.tar.bz2
+    tar -xjf gmp-6.0.0a.tar.bz2;
+    cd gmp-6.0.0 && ./configure --prefix=$SYMENGINE_INSTALL_DIR --enable-cxx && make -j8 install && cd ..;
+
+  else
+    export CC="gcc" && export CXX="g++-4.7" && export CMAKE_C_COMPILER=$CC && export CMAKE_CXX_COMPILER=$CXX  
+  fi
+  cd $HOME/symengine
+  # cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=$SYMENGINE_INSTALL_DIR/ -DBUILD_TESTS=no  -DBUILD_SHARED_LIBS:BOOL=ON .
+  cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=$SYMENGINE_INSTALL_DIR/ -DBUILD_TESTS=no  .
+  make
+  make install
+  echo "listing symengine install dir" && echo "======================"
```

Comment:
Instead of lines 187 to 199, you can do 
```
- export TEST_CPP="no"
- cd $HOME/symengine
- source bin/install_travis.sh
- bin/test_travis.sh
```
which uses the script from SymEngine C++ repo. This will install symengine to `$our_install_dir`